src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs

```diff
@@ -580,8 +580,14 @@ internal void TogglePreviewControlAllowance()
 
             if (previewEnabled == false && !PreviewControl.StaysOpen)
             {
-                PreviewControl.TransitionToState(PreviewControl.State.Condensed);
-                PreviewControl.TransitionToState(PreviewControl.State.Hidden);
+                if (PreviewControl.IsExpanded)
+                {
+                    PreviewControl.TransitionToState(PreviewControl.State.Condensed);
```

Comment:
Ok. 
FYI : It is not right to transition like this. There has to be an intervention either externally or self.
but for now merge this in.